spec/datadog/tracing/contrib/aws/instrumentation_spec.rb

```diff
@@ -74,7 +74,7 @@
     end
   end
 
-  context 'with an S3 client' do
+  context 'with an S3 client' if: RUBY_VERSION >= '2.2.0' do
```

Comment:
Unfortunately, there are flaky specs for aws 2.x need to be addressed. 

I was able to reproduce it after changing dependency version with

`bundle exec appraisal ruby-3.1.2-contrib rspec spec/datadog/tracing/contrib/aws/instrumentation_spec.rb spec/datadog/tracing/contrib/aws/parsed_context_spec.rb --seed 1331`

Reply:
@TonyCTHsu 
Im not sure I understand, yes the tests were failing on ruby 3.1.2 but according to this[ aws doc](https://aws.amazon.com/blogs/developer/announcing-the-end-of-support-for-ruby-runtimes-1-9-2-0-2-1-and-2-2-for-the-aws-sdk-for-ruby/#:~:text=Ruby%201.9.3%20%E2%80%93%20EOL,2018%2D03%2D31)
, ruby 2.1 and 2.2 are not supported in the aws sdk anymore thus the reason why I added `if: RUBY_VERSION >= '2.2.0'`. These tests use the aws sdk thus I added the check to only run them if the ruby version is 2.3.0 or greater. 

All checks passed in CI now

Reply:
please correct me if i'm wrong but I believe this does have to do with older rubies not being compatible with the aws-sdk thus for the unit tests we should accept 2.2.0 and greater

Reply:
>I believe this does have to do with older rubies not being compatible with the aws-sdk

No, it is due to the fact that other ruby versions appraisal groups have aws 3.x installed as dependency. You can verified the behaviour by installing aws 2.x